cmd/geth/convert_block_firehose_to_geth_test.go

```diff
@@ -0,0 +1,146 @@
+					WithdrawalsRoot:  []byte{101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132},
+					BlobGasUsed:      new(uint64),
+					ExcessBlobGas:    new(uint64),
+					ParentBeaconRoot: []byte{201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232},
+					RequestsHash:     []byte{151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182},
+				},
+				TransactionTraces: []*pbeth.TransactionTrace{
+					{
+						Type:     0, // LegacyTx
+						Hash:     []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32},
+						Nonce:    0,
+						GasPrice: &pbeth.BigInt{Bytes: big.NewInt(20000000000).Bytes()},
+						GasLimit: 21000,
+						To:       []byte{11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30},
+						Value:    &pbeth.BigInt{Bytes: big.NewInt(1000000000000000000).Bytes()},
+						Input:    []byte{},
+						V:        []byte{27},
+						R:        []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32},
+						S:        []byte{33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64},
+						Status:   1,
+						GasUsed:  21000,
+						Receipt: &pbeth.TransactionReceipt{
+							StateRoot:         []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32},
+							CumulativeGasUsed: 21000,
+							LogsBloom:         make([]byte, 256), // Bloom filter is 256 bytes
+							Logs:              []*pbeth.Log{},
+						},
+						MaxPriorityFeePerGas:  &pbeth.BigInt{Bytes: []byte{0}},
+						MaxFeePerGas:          &pbeth.BigInt{Bytes: []byte{0}},
+						AccessList:            []*pbeth.AccessTuple{},
+						BlobGasFeeCap:         &pbeth.BigInt{Bytes: []byte{0}},
+						BlobHashes:            [][]byte{},
+						SetCodeAuthorizations: []*pbeth.SetCodeAuthorization{},
+					},
+				},
+				Uncles: []*pbeth.BlockHeader{},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			block, err := convertFirehoseBlockToGethBlock(tt.pbBlock, big.NewInt(0))
+			if (err != nil) != tt.wantErr {
+				t.Errorf("convertFirehoseBlockToGethBlock() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !tt.wantErr && block == nil {
+				t.Error("convertFirehoseBlockToGethBlock() returned nil block when no error expected")
+			}
+
+			// Only do golden file comparison for the valid block case
+			if tt.name == "valid block" && !tt.wantErr {
```

Comment:
Do we really need to check for a specific test name? That is error prone, dont just `wantErr` is enough here?